packages/viz/src/lib/Layer.ts

```diff
@@ -1,107 +1,107 @@
-import { Credentials, defaultCredentials } from '@carto/toolkit-core';
-import { MapInstance, MapOptions, Maps } from '@carto/toolkit-maps';
 import { MVTLayer } from '@deck.gl/geo-layers';
 
-import Source from './Source';
-import {defaultStyles, Style} from './style';
+import { CARTOSource } from './sources/CARTOSource';
+import { Source } from './sources/Source';
 
-const defaultMapOptions: MapOptions = {
-  vector_extent: 2048,
-  vector_simplify_extent: 2048,
-  metadata: {
-    geometryType: true
-  }
-};
+import {defaultStyles, Style} from './style';
 
 export class Layer {
-  private _mapsClientInstance: Maps;
-  private _credentials: Credentials;
 
-  private _layerSource: Source;
-  private _layerStyles: Style;
-  private _layerOptions: MapOptions;
-  private _layerInstantiation: Promise<MapInstance>;
+  private _source: Source;
+  private _styles: Style;
 
   private _deckInstance: any;
-  private _mvtLayerInstance: MVTLayer<string> | undefined;
-
-  constructor(source: string, styles = {}, options: LayerOptions = {}) {
-    const { mapOptions = {}, credentials = defaultCredentials } = options;
-
-    this._credentials = credentials;
-    this._mapsClientInstance = new Maps(this._credentials);
+  private _deckLayer: any;
```

Comment:
It's not a generic Deck Layer, because it could have MVTLayer for CARTOSource, DataObservatoryLayer for DataObservatorySource and GeoJSONLayer in the future. 

There is an issue with the types. I've tried to add TileLayer, MVTLayer and Layer, but the typing complains because the getPickinfo are differents.

TileLayer getPickingInfo:
```
getPickingInfo({ info, sourceLayer }: {
			info: any;
			sourceLayer: any;
		}): any;
```

Layer: 
```
	getPickingInfo({ info, mode }: {
			info: any;
			mode: any;
		}): any;
```

CompositeLayer:
```
	getPickingInfo({ info }: {
			info: any;
		}): any;
```